docs/interface.md

```diff
@@ -0,0 +1,56 @@
+# The Interface
+
+All interactions with the Exercism website are handled automatically. Test runners have the single responsibility of taking a solution, running its tests and returning a standardised output.
+
+## Execution
+
+- A test runner should provide an executable script. You can find more information in the [docker.md](https://github.com/exercism/automated-mentoring-support/blob/master/docs/docker.md) file.
+- The script will receive two parameters:
+  - The slug of the exercise (e.g. `two-fer`).
+  - A path to a directory containing the submitted file(s) (with a trailing slash).
+- The script must write a file to that directory named `report.json`
+
+## Output format
+
+The `report.json` file should be structured as followed:
+
+```json
+{
+  "status": "fail",
+  "tests": [
+    {
+      "name": "Test that the thing works",
+      "status": "fail",
+      "expected": "42",
```

Comment:
Having checked the dotnet test output and a JUnit sample file, it looks like the `expected` and `actual` values are not something that is returned. The assertion data (expected/actual) are contained in the message. Is perhaps having only the message and not the  `expected` and `actual` values also sufficient?

Reply:
Good point! Now that you mention it, those values aren't returned by Rust either. At the same time, I suspect that it's probably safe to omit them; what the student or mentor needs to know is that a test failed; it should be relatively simple from there to track down the problem.

Reply:
We want to show a diff in the UI if i understand correctly. The format of the junit message is generally the same, so these values can be extracted.

We probably need the case where it can't tho and just show a message.

Reply:
They're not returned in Julia either. `expected` would always be `true` because the tests check if an expression evaluates to true, not if an expected value matches the returned value. For simple tests like `f(foo, bar) == expected` this could be extracted easily but many tests don't follow that format.

Reply:
Same in the Go output. Currently it is printed as a manually generated string:

```go
t.Errorf("Score(%q) expected %d, Actual %d", test.input, test.expected, actual)
```

A special output could be used here to parse the expected and actual values from. Probably requires checking and touching all tests for all exercises.

Multi-return values also might need some thought.

Reply:
I wonder if in the end it would be easier to implement AND easier to read for the students to see the familiar test output they would get from running the tests locally.

Reply:
FWIW: most test frameworks _start_ with only having `assert` and later adding `assertEquals` because `assert` just gives so _little_ information. If your current tests only use `assert`, the expected value would be `"true"` and the actual value would be `"false"`.

"test failed because assertion failed" usually doesn't give _enough_ information to _why_ it failed. Adding "test failed because expected 50, actual 49" instantly tells you _why_ it failed.

Reply:
> I wonder if in the end it would be easier to implement AND easier to read for the students/mentors to see the familiar test output they would get from running the tests locally.

This is what I'm in favor of.

> "test failed because expected 50, actual 49" 

Yep, this is much better. Virtually all the C# tests are defined using the specialized `Assert.Equal` assertion, which error message will include the expected and actual value.

Reply:
Strongly agreed with @tehsphinx . In my perspective, for different languages and different test frameworks, it would bring best experience for users to see what they see at local, in that:
- It brings seamless experience between local and website, adding no mental burdens;
- The output from each test framework describes their test best; 

@iHiD i might be thinking too wild: instead of emitting a JSON, is it possible to capture the Stdout & Stderr and showing that? Of course have to filter them or giving a certain lower level of verbosity.  And addition to this output, could add a `summarizer()` interface, that can be implemented specifically for each test framework, taking the output of test as input, and gives the JSON format as output to generate a summary list.